### ISSUE TITLE
iDrive e2 fails to load images

### ISSUE THREAD

**Issue by thepaperpilot** [babybuddy/babybuddy#733]
I'm using an s3 compatible api (iDrive e2) and images upload successfully but don't load at all. If I go to the URL directly I get an xml response saying its a SignatureDoesNotMatch error.

If this is non-trivial to fix, one workaround that should work is making the bucket public. However that gives me a separate endpoint that won't have write permissions (and ofc I'm not a fan of making my baby's images accessible without auth), and there's no documented way to use a different endpoint for uploading versus rendering pages.

**Comment by cdubz** [comment]
Hm on the backend Baby Buddy uses https://django-storages.readthedocs.io/en/latest/ for file storage support. I don't see iDrive mentioned anywhere in their documentation or issues. Nor do I see anything about "S3 compatible" service support.

This the only potential relevant info I could find but seems pretty light... https://stackoverflow.com/questions/71302380/using-non-aws-s3-storage-with-django

**Comment by thepaperpilot** [comment]
"s3 compatible" is a pretty common term when looking into object storage. Django may not mention it specifically but [baby buddy does](https://docs.baby-buddy.net/configuration/storage/). 

I'd already been including the scheme in the endpoint url since the docs said to, so that SO post didn't help me unfortunately.

**Comment by cdubz** [comment]
Ha yes the documentation does say that I bet it’s never actually been tested… 😬

Does iDrive have a free tier I could fiddle with?

**Comment by thepaperpilot** [comment]
Yes, so long as you don't pay for an entire year, you get 10GB free. ([pricing](https://www.idrive.com/object-storage-e2/pricing))
